synapse/handlers/account_validity.py

```diff
@@ -429,7 +428,10 @@ async def renew_account_for_user(
         # Check if renewed users should be reintroduced to the user directory
         if self._show_users_in_user_directory:
             # Show the user in the directory again by setting them to active
-            await self.profile_handler.set_active(
+            # We get the profile handler here so that we don't cause a cyclic dependency
+            # at startup.
+            # FIXME: this doesn't work with the new account validity module stuff
+            await self.hs.get_profile_handler().set_active(
                 [UserID.from_string(user_id)], True, True
             )
```

Comment:
In what way does it not work? Just this call or setting profiles to active in general?

Reply:
I've expanded that comment in https://github.com/matrix-org/synapse-dinsic/pull/102/commits/a45de9e8d01266fac367a5a487271b383116aaf7